src/server/utils/generate-manifest-json.ts

```diff
@@ -1,19 +1,43 @@
-import { readFile } from "fs/promises";
 import { Site } from "lemmy-js-client";
-import path from "path";
 import { fetchIconPng } from "./fetch-icon-png";
+import { getStaticDir } from "@utils/env";
 
+type Icon = { sizes: string; src: string; type: string; purpose: string };
 const iconSizes = [72, 96, 128, 144, 152, 192, 384, 512];
-
-const defaultLogoPathDirectory = path.join(
-  process.cwd(),
-  "dist",
-  "assets",
-  "icons",
-);
+let icons: Icon[] | null = null;
 
 export default async function (site: Site) {
-  const icon = site.icon ? await fetchIconPng(site.icon) : null;
+  if (!icons) {
+    try {
+      const icon = site.icon ? await fetchIconPng(site.icon) : null;
+
+      icons = await Promise.all(
```

Comment:
it still looks like the icons array is unassigned if this call fails? so it runs again every time?